### ISSUE TITLE
Visual Studio Errors while installing karma

### ISSUE THREAD

**Issue by op1ekun** [karma-runner/karma#1691]
Hi guys,

I'm running `npm install karma --save-dev` from my git bash and this is what I get some message about building a solution (?):

```
C:\workspace\npmTest\node_modules\karma\node_modules\socket.io\node_modules\engine.io\node_modules\ws\node_modules\utf-8-validate>if not defined npm_config_node_gyp (node "C:\bin\nodejs\node_modules\npm\bin\node-gyp-bin\\..\..\node_modules\node-gyp\bin\node-gyp.js" rebuild )  else (node  rebuild )
Building the projects in this solution one at a time. To enable parallel build, please add the "/m" switch.
```
followed by a list of errors:
```
C:\Program Files (x86)\MSBuild\Microsoft.Cpp\v4.0\V140\Microsoft.CppBuild.targets(366,5): 
warning MSB8003: Could not find WindowsSDKDir variable f rom the registry. 
TargetFrameworkVersion or PlatformToolset may be set to an invalid version number. [C:\workspace\npmTest\node_modules\karma\node_modules\socket.io\node_modules\engine.io\node_modules\ws\node_modules\bufferutil\build\bufferutil.vcxproj]

TRACKER : error TRK0005: Failed to locate: "CL.exe". 
The system cannot find the file specified. [C:\workspace\npmTest\node_modules\karma\node_modu les\socket.io\node_modules\engine.io\node_modules\ws\node_modules\bufferutil\build\bufferutil.vcxproj]

gyp ERR! build error
gyp ERR! stack Error: `C:\Program Files (x86)\MSBuild\14.0\bin\msbuild.exe` failed with exit code: 1
gyp ERR! stack     at ChildProcess.onExit (C:\bin\nodejs\node_modules\npm\node_modules\node-gyp\lib\build.js:270:23)
gyp ERR! stack     at emitTwo (events.js:87:13)
gyp ERR! stack     at ChildProcess.emit (events.js:172:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:200:12)
gyp ERR! System Windows_NT 10.0.10240
gyp ERR! command "C:\\bin\\nodejs\\node.exe" "C:\\bin\\nodejs\\node_modules\\npm\\node_modules\\node-gyp\\bin\\node-gyp.js" "rebuild"
gyp ERR! cwd C:\workspace\npmTest\node_modules\karma\node_modules\socket.io\node_modules\engine.io\node_modules\ws\node_modules\utf-8-validate
gyp ERR! node -v v4.2.0
gyp ERR! node-gyp -v v3.0.3
gyp ERR! not ok
```

I have just installed Visual Studio Community 2015, but what it has to do with a NPM PACKAGE?
I couldn't find anything about this particular issue in the web.

Fortunately Karma seems to run fine, but this is very confusing.

Env:
Windows 10
git bash
node.js 4.2.0
Karma 0.13.15

**Comment by op1ekun** [comment]
I'm still puzzled why Karma started to build stuff using Visual Studio etc.

However it seems that when I installed my version of Visual Studio Community 2015 I only installed it with default Web Development tools.

I have just installed Visual C++ tools package, and voila! The installation finishes with no errors.

I hope one day this will help someone else :)

**Comment by karmarunnerbot** [comment]
thanks for this!

On Wed, Nov 4, 2015 at 10:51 AM, Lukasz Piatkowski <notifications@github.com

**Comment by MoffDo** [comment]
Hello and THANK YOU!

Here's what I had to do to solve my problem:

I tried to install Visual C++ tools package, but I couldn't due to "important updates".
After restarting the system and installing those updates, I could install, as you said, Visual C++ tools package from my Visual Studio Community 2015.
(Attempting to open a new Visual C++ project forces you to install it, because that package is not installed by default)

After the installation, I tried to run 'grunt test' (using Karma) on my Yeoman project workspace using Angular generator.

However phantomjs was not installed properly, so I reinstalled it using:
npm install phantomjs

Then I tried to run the grunt test and it worked successfully!!

-- Done, without errors. --

I'm so glad you posted this! Thank you very very much!
